theano/scan_module/scan_op.py

```diff
@@ -2198,7 +2198,7 @@ def compute_all_gradients(known_grads):
                 dC_dinps_t[dx + self.n_seqs] += dC_dXtm1
         # Construct scan op
         # Seqs
-        outer_inp_seqs = [x[::-1] for x in inputs[1:1 + self.n_seqs]]
+        outer_inp_seqs = [x[:grad_steps][::-1] for x in inputs[1:1 + self.n_seqs]]
```

Comment:
It would not work if `grad_steps <= 0`, though. Are we sure it is never going to be 0?
```python
>>> range(10)[0-1::-1]
[9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
>>> range(10)[:0][::-1]
[]
```